service/type.go

```diff
@@ -0,0 +1,97 @@
+package service
+
+import "github.com/mesg-foundation/engine/hash"
+
+// MainServiceKey is key for main service.
+const MainServiceKey = "service"
+
+// StatusType of the service.
+type StatusType uint
+
+// Possible statuses for service.
+const (
+	UNKNOWN StatusType = iota
+	STOPPED
+	STARTING
+	PARTIAL
+	RUNNING
+)
+
+func (s StatusType) String() string {
+	switch s {
+	case STOPPED:
+		return "STOPPED"
+	case STARTING:
+		return "STARTING"
+	case PARTIAL:
+		return "PARTIAL"
+	case RUNNING:
+		return "RUNNING"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// WARNING about hash tags on Service type and its inner types:
+// * never change the name attr of hash tag. use an incremented value for
+// name attr when a new configuration field added to Service.
+// * don't increment the value of name attr if corresponding field's name
+// changed but its behavior remains the same.
+// * this is required for not breaking Service IDs unless there is a behavioral
+// change.
+
+// Service represents a MESG service.
+type Service struct {
+	Hash          hash.Hash     `hash:"-" validate:"required"`                                // Hash is calculated from the combination of service's source and mesg.yml. It represents the service uniquely.
+	Sid           string        `hash:"name:1"  validate:"required,printascii,max=63,domain"` // Sid is the service id. It needs to be unique and can be used to access to service.
+	Name          string        `hash:"name:2" validate:"required,printascii"`                // Name is the service name.
+	Description   string        `hash:"name:3" validate:"printascii"`                         // Description is service description.
+	Tasks         []*Task       `hash:"name:4" validate:"dive,required"`                      // Tasks are the list of tasks that service can execute.
+	Events        []*Event      `hash:"name:5" validate:"dive,required"`                      // Events are the list of events that service can emit.
+	Dependencies  []*Dependency `hash:"name:6" validate:"dive,required"`                      // Dependencies are the Docker containers that service can depend on.
+	Configuration *Dependency   `hash:"name:8" validate:"required"`                           // Configuration of the service
+	Repository    string        `hash:"name:7" validate:"omitempty,uri"`                      // Repository holds the service's repository url if it's living on a Git host.
+	Source        string        `hash:"name:9" validate:"required,printascii"`                // Source is the hash id of service's source code on IPFS.
+}
+
+// Dependency represents a Docker container and it holds instructions about
+// how it should run.
+type Dependency struct {
+	Key         string   `hash:"1" validate:"printascii"`                  // Key is the key of dependency.
```

Comment:
flag name is missing... it was already missing..
```suggestion
	Key         string   `hash:"name:1" validate:"printascii"`                  // Key is the key of dependency.
```

Reply:
done in https://github.com/mesg-foundation/engine/pull/1186